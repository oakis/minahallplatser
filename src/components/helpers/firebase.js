import fetch from 'react-native-cancelable-fetch';
import { firebaseFunctionsUrl } from '../../Server';
import { handleJsonFetch } from './';

export const updateDeparturesCount = (count) => {
    if (__DEV__) {
        return;
    }
    const url = `${firebaseFunctionsUrl}/addDeparturesCount?count=${count}`;
    fetch(url, {}, 'updateDeparturesCount')
    .finally(handleJsonFetch)
    .then(({ message }) => {
        window.log(`updateDeparturesCount(): OK (${message})`);
    })
    .catch((err) => {
        window.log(`updateDeparturesCount(): FAILED' ${message})`);
        new Error(err);
    });
};

export const getDeparturesCount = () => {
    const url = `${firebaseFunctionsUrl}/getDeparturesCount`;
    return fetch(url, {}, 'getDeparturesCount')
    .finally(handleJsonFetch)
    .then(({ departuresCount }) => {
        window.log('getDeparturesCount(): OK', departuresCount);
        return departuresCount;
    })
    .catch((err) => {
        window.log('getDeparturesCount(): FAILED', err);
        new Error(err);
    });
};

export const updateStopsCount = () => {
    if (__DEV__) {
        return;
    }
    const url = `${firebaseFunctionsUrl}/addStopsCount`;
    fetch(url, {}, 'updateStopsCount')
    .finally(handleJsonFetch)
    .then(({ message }) => {
        window.log('updateStopsCount(): OK', message);
    })
    .catch((err) => {
        window.log('updateStopsCount(): FAILED', message);
        new Error(err);
    });
};

export const getStopsCount = () => {
    const url = `${firebaseFunctionsUrl}/getStopsCount`;
    return fetch(url, {}, 'getStopsCount')
    .finally(handleJsonFetch)
    .then(({ stopsCount }) => {
        window.log('getStopsCount(): OK', stopsCount);
        return stopsCount;
    })
    .catch((err) => {
        window.log('getStopsCount(): FAILED', err);
        new Error(err);
    });
};

export const incrementStopsOpened = (user, stopId) => {
    const url = `${firebaseFunctionsUrl}/incrementStopsOpen?user=${user}&stopId=${stopId}`;
    fetch(url, {}, 'incrementStopsOpened')
    .finally(handleJsonFetch)
    .then(({ message }) => {
        window.log('incrementStopsOpened(): OK', message);
    })
    .catch((err) => {
        window.log('incrementStopsOpened(): FAILED', err);
        new Error(err);
    });
}