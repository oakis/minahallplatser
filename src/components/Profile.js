import React, { PureComponent } from 'react';
import {
    View,
    ScrollView,
} from 'react-native';
import firebase from 'react-native-firebase';
import {
    ListItem,
    ListItemSeparator,
} from './common';

class Profile extends PureComponent {
    render() {
        const user = firebase.auth().currentUser;
        console.log(user);
        return (
            <View style={{ flex: 1 }}>
                <ScrollView scrollEnabled keyboardShouldPersistTaps="always">
                    {user.providerData[0].displayName ?
                        <View>
                            <ListItem
                                text={user.providerData[0].displayName}
                                avatar={user.providerData[0].photoURL}
                                iconVisible
                            />
                            <ListItemSeparator />
                        </View>
                    : null}
                    <ListItem
                        text={user.email}
                        icon="ios-mail"
                        iconVisible
                    />
                    <ListItemSeparator />
                </ScrollView>
            </View>
        );
    }
}

export default Profile;
