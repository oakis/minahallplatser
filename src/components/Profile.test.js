import React from 'react';
import { shallow } from 'enzyme';
import firebase from 'react-native-firebase';
import Profile from './Profile';

it('should match snapshot if e-mail account', () => {
    const wrapper = shallow(<Profile />);
    expect(wrapper).toMatchSnapshot();
});

it('should match snapshot if facebook account', () => {
    firebase.auth().currentUser.providerData.push({ displayName: 'John Doe' });
    const wrapper = shallow(<Profile />);
    expect(wrapper).toMatchSnapshot();
});
