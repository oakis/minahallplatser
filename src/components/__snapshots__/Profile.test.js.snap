// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot if e-mail account 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ScrollView
        keyboardShouldPersistTaps="always"
        scrollEnabled={true}
      >
        <ListItem
          icon="ios-mail"
          iconVisible={true}
          text={undefined}
        />
        <ListItemSeparator />
      </ScrollView>,
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          null,
          <ListItem
            icon="ios-mail"
            iconVisible={true}
            text={undefined}
          />,
          <ListItemSeparator />,
        ],
        "keyboardShouldPersistTaps": "always",
        "scrollEnabled": true,
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "ios-mail",
            "iconVisible": true,
            "text": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ScrollView
          keyboardShouldPersistTaps="always"
          scrollEnabled={true}
        >
          <ListItem
            icon="ios-mail"
            iconVisible={true}
            text={undefined}
          />
          <ListItemSeparator />
        </ScrollView>,
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            null,
            <ListItem
              icon="ios-mail"
              iconVisible={true}
              text={undefined}
            />,
            <ListItemSeparator />,
          ],
          "keyboardShouldPersistTaps": "always",
          "scrollEnabled": true,
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "icon": "ios-mail",
              "iconVisible": true,
              "text": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should match snapshot if facebook account 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ScrollView
        keyboardShouldPersistTaps="always"
        scrollEnabled={true}
      >
        <View>
          <ListItem
            avatar={undefined}
            iconVisible={true}
            text="John Doe"
          />
          <ListItemSeparator />
        </View>
        <ListItem
          icon="ios-mail"
          iconVisible={true}
          text={undefined}
        />
        <ListItemSeparator />
      </ScrollView>,
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <View>
            <ListItem
              avatar={undefined}
              iconVisible={true}
              text="John Doe"
            />
            <ListItemSeparator />
          </View>,
          <ListItem
            icon="ios-mail"
            iconVisible={true}
            text={undefined}
          />,
          <ListItemSeparator />,
        ],
        "keyboardShouldPersistTaps": "always",
        "scrollEnabled": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ListItem
                avatar={undefined}
                iconVisible={true}
                text="John Doe"
              />,
              <ListItemSeparator />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "avatar": undefined,
                "iconVisible": true,
                "text": "John Doe",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "ios-mail",
            "iconVisible": true,
            "text": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ScrollView
          keyboardShouldPersistTaps="always"
          scrollEnabled={true}
        >
          <View>
            <ListItem
              avatar={undefined}
              iconVisible={true}
              text="John Doe"
            />
            <ListItemSeparator />
          </View>
          <ListItem
            icon="ios-mail"
            iconVisible={true}
            text={undefined}
          />
          <ListItemSeparator />
        </ScrollView>,
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <View>
              <ListItem
                avatar={undefined}
                iconVisible={true}
                text="John Doe"
              />
              <ListItemSeparator />
            </View>,
            <ListItem
              icon="ios-mail"
              iconVisible={true}
              text={undefined}
            />,
            <ListItemSeparator />,
          ],
          "keyboardShouldPersistTaps": "always",
          "scrollEnabled": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ListItem
                  avatar={undefined}
                  iconVisible={true}
                  text="John Doe"
                />,
                <ListItemSeparator />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "avatar": undefined,
                  "iconVisible": true,
                  "text": "John Doe",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "icon": "ios-mail",
              "iconVisible": true,
              "text": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
